common/types/message/message.go

```diff
@@ -57,106 +39,25 @@ const (
 )
 
-// GenerateToken generates token
-func GenerateToken() (string, error) {
-	b := make([]byte, 16)
-	if _, err := rand.Read(b); err != nil {
-		return "", err
-	}
-	return hex.EncodeToString(b), nil
-}
-
-// ProofMsg is the data structure sent to the coordinator.
-type ProofMsg struct {
-	*ProofDetail `json:"zkProof"`
-	// Prover signature
-	Signature string `json:"signature"`
-
-	// Prover public key
-	publicKey string
-}
-
-// Sign signs the ProofMsg.
-func (a *ProofMsg) Sign(priv *ecdsa.PrivateKey) error {
-	hash, err := a.ProofDetail.Hash()
-	if err != nil {
-		return err
-	}
-	sig, err := crypto.Sign(hash, priv)
-	if err != nil {
-		return err
-	}
-	a.Signature = hexutil.Encode(sig)
-	return nil
-}
-
-// Verify verifies ProofMsg.Signature.
-func (a *ProofMsg) Verify() (bool, error) {
-	hash, err := a.ProofDetail.Hash()
-	if err != nil {
-		return false, err
-	}
-	sig := common.FromHex(a.Signature)
-	// recover public key
-	if a.publicKey == "" {
-		pk, err := crypto.SigToPub(hash, sig)
-		if err != nil {
-			return false, err
-		}
-		a.publicKey = common.Bytes2Hex(crypto.CompressPubkey(pk))
-	}
-
-	return crypto.VerifySignature(common.FromHex(a.publicKey), hash, sig[:len(sig)-1]), nil
-}
-
-// PublicKey return public key from signature
-func (a *ProofMsg) PublicKey() (string, error) {
-	if a.publicKey == "" {
-		hash, err := a.ProofDetail.Hash()
-		if err != nil {
-			return "", err
-		}
-		sig := common.FromHex(a.Signature)
-		// recover public key
-		pk, err := crypto.SigToPub(hash, sig)
-		if err != nil {
-			return "", err
-		}
-		a.publicKey = common.Bytes2Hex(crypto.CompressPubkey(pk))
-		return a.publicKey, nil
-	}
-
-	return a.publicKey, nil
-}
-
-// TaskMsg is a wrapper type around db ProveTask type.
-type TaskMsg struct {
-	UUID             string            `json:"uuid"`
-	ID               string            `json:"id"`
-	Type             ProofType         `json:"type,omitempty"`
-	ChunkTaskDetail  *ChunkTaskDetail  `json:"chunk_task_detail,omitempty"`
-	BatchTaskDetail  *BatchTaskDetail  `json:"batch_task_detail,omitempty"`
-	BundleTaskDetail *BundleTaskDetail `json:"bundle_task_detail,omitempty"`
-}
-
 // ChunkTaskDetail is a type containing ChunkTask detail.
 type ChunkTaskDetail struct {
 	BlockHashes []common.Hash `json:"block_hashes"`
 }
 
 // BatchTaskDetail is a type containing BatchTask detail.
 type BatchTaskDetail struct {
-	ChunkInfos      []*ChunkInfo     `json:"chunk_infos"`
-	ChunkProofs     []*ChunkProof    `json:"chunk_proofs"`
-	ParentStateRoot common.Hash      `json:"parent_state_root"`
-	ParentBatchHash common.Hash      `json:"parent_batch_hash"`
-	BatchHeader     *codecv3.DABatch `json:"batch_header"`
+	ChunkInfos      []*ChunkInfo  `json:"chunk_infos"`
+	ChunkProofs     []*ChunkProof `json:"chunk_proofs"`
+	ParentStateRoot common.Hash   `json:"parent_state_root"`
+	ParentBatchHash common.Hash   `json:"parent_batch_hash"`
+	// FIXME: this structure does not support json marshal/unmarshal well.
```

Comment:
@roynalnaruto this structure does not support `json` marshal/unmarshal well, we can find some ways to fix this.

Reply:
tried JSON marshal/unmarshal, looks good.
```
{"Version":3,"BatchIndex":0,"L1MessagePopped":0,"TotalL1MessagePopped":0,"DataHash":"0x9f81f6879f121da5b7a37535cdb21b3d53099266de57b1fdf603ce32100ed541","BlobVersionedHash":"0x01bbc6b98d7d3783730b6208afac839ad37dcf211b9d9e7c83a5f9d02125ddd7","ParentBatchHash":"0x0000000000000000000000000000000000000000000000000000000000000000","LastBlockTimestamp":1669364522,"BlobDataProof":[9,143,31,19,111,87,52,3,152,24,190,227,82,34,211,90,150,172,215,209,113,32,206,136,22,48,117,39,209,155,173,234,23,208,19,190,94,246,150,207,188,5,185,123,179,34,165,135,67,44,44,178,60,72,72,212,215,203,132,83,196,117,179,141]}
```